webapp/src/templates/directives/header.html

```diff
@@ -115,42 +115,20 @@
     </div>
 
     <div class="tabs small-font">
-
-      <a ui-sref="messages.detail" id="messages-tab" class="messages-tab" ng-class="{'selected': headerCtrl.currentTab === 'messages'}" mm-auth="can_view_messages,can_view_messages_tab">
-        <div class="mm-icon" ng-class="{'mm-icon-inverse': headerCtrl.currentTab !== 'messages'}">
-          <span class="fa fa-envelope"></span>
-          <span class="mm-badge-overlay mm-badge-overlay-top" ng-hide="headerCtrl.unreadCount.message <= 0">
+      <a ng-repeat="tab in headerCtrl.filteredTabs() track by tab.name" ui-sref="{{tab.state}}" ng-attr-class="{'analytics-tab' : tab.name === 'analytics'}" id="{{tab.name}}-tab" class="{{tab.name}}-tab" ng-class="{'selected': headerCtrl.currentTab === {{tab.name}}}">
+        <div class="mm-icon" ng-class="{'mm-icon-inverse': headerCtrl.currentTab !== {{tab.name}}}">
+          <span class="fa {{tab.icon}}"></span>
+          <span ng-if="tab.name === 'messages'" class="mm-badge-overlay mm-badge-overlay-top" ng-hide="headerCtrl.unreadCount.message <= 0">
```

Comment:
I think you should be able to combine the two counting fragments here, into block that generically displays a count if that tab has one.

Have a think about what knowledge you have at your disposable (what values exist which make a tab with a count different from another tab), and see how you go. Ping me on slack if you want to talk about it.